flume-ng-core/src/main/java/org/apache/flume/source/http/HTTPSource.java

```diff
@@ -130,11 +131,13 @@ public void configure(Context context) {
 
       if (sslEnabled) {
         LOG.debug("SSL configuration enabled");
-        keyStorePath = context.getString(HTTPSourceConfigurationConstants.SSL_KEYSTORE);
+        keyStorePath = context.getString(HTTPSourceConfigurationConstants.SSL_KEYSTORE,
+                SSLUtil.getGlobalKeystorePath());
         Preconditions.checkArgument(keyStorePath != null && !keyStorePath.isEmpty(),
                                     "Keystore is required for SSL Conifguration" );
-        keyStorePassword =
-            context.getString(HTTPSourceConfigurationConstants.SSL_KEYSTORE_PASSWORD);
+        keyStorePassword = context.getString(
+                HTTPSourceConfigurationConstants.SSL_KEYSTORE_PASSWORD,
+                SSLUtil.getGlobalKeystorePassword());
```

Comment:
We should also capture keystore type here, really. That can be added in L207

Reply:
Good question. HTTPSource does not support the type parameter in its config explicitly, so I did not add it here. We can add it though.

Reply:
I think we should add it in for consistency (sorry, I know that's extra work)

Reply:
@tmgstevens I have a change on the way where I have added SSL/TLS support for the syslog source. I have made some refactoring that will add this parameter and makes SSL/TLS handling unified. So if it is ok for you I would commit this and add the keystore type with that change.

Reply:
I agree with @szaboferee, let's go ahead with committing this PR.